doc/rds-day-2.md

```diff
@@ -0,0 +1,153 @@
+1. Define a scheduled maintenance window.
+2. Queue changes that you want to make.
+3. During the next scheduled maintenance window, changes are applied.
+
+This approach is fundamentally different from GitOps. In GitOps, when a change is checked into your repository, it is expected that actual resources are to match the specifications provided in the repository. 
+
+RDS supports applying these changes immediately instead of waiting for a scheduled maintenance window, and when using Crossplane AWS providers, they have the option to apply changes immediately as well. This is the option that should be used when using RDS with GitOps. However this leads to problems when enabling self service model where developers can provision resources on their own. 
+
+One of most notable problems is that updates made to certain fields could trigger database restarts. Developers may not know which fields would cause restarts because they are not familiar with underlying technologies. You could document potentially dangerous fields, but it is not enough to reliably stop it from happening. 
+
+
+## Parameter Groups
+[Parameter Groups](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_WorkingWithParamGroups.html) define how the underlying database engine is configured. For example you may wish to change the `binlog_cache_size` configuration value for your MySQL database. 
+
+In Parameter Groups, there are two types of parameters: dynamic and static.
+Dynamic parameters do not require a restart for their values to be applied to the running instance / cluster.  Static parameters require a restart for their values to be applied. Additionally, dynamic parameters support specifying how changes to them are applied. When `immediate` is specified the changes to dynamic parameters are applied immediately. When `pending-reboot` is specified, the changes to dynamic parameters are applied during next restart or during the next maintenance window, whichever is earlier. 
+
+Since static parameters do not support `immediate` apply option, specifying this in your composition could lead to some unexpected error. Therefore, extra care should be taken when exposing this resource to your end users. End users may not be aware of underlying engine specifications.
+
+Summarizing everything above effectively means there are a few general approaches to managing RDS configuration changes. 
+1. You want to ensure that parameter group values in running cluster / instance match what is defined in your Git repository with no delay. The only sure way to ensure that is restarting the cluster/ instance during the reconciliation process. 
+3. You can wait for parameter group changes to be applied during the next maintenance window. This means you may need to wait maximum 7 days for the changes to be applied. 
```

Comment:
not sure how this can be handled effectively via GitOps? is it that the request to make the change keeps getting applied to no avail? if so, we should discuss it. also we should mention that they need to provide out-of-band mechanisms to track this.

Reply:
Changes to parameter groups will say they are synced, and it is technically correct because parameter group is its own concept and multiple instances can share a parameter group. It's usually not what users actually want though. They want the changes to parameter groups to be applied to instances, and to do that, you need to restart instances that are using the modified parameter groups.

This is communicated via the `ParameterApplyStatus` field. When you modify a static parameter in a parameter group, this is what you get when you do a describe instance. 
```bash
aws rds describe-db-instances
      {
            "DBInstanceIdentifier": "test1-instance-1",
            "DBInstanceClass": "db.t3.small",
            "Engine": "aurora-mysql",
            "DBInstanceStatus": "available",
            "MasterUsername": "admin",
            "DBParameterGroups": [
                {
                    "DBParameterGroupName": "test-instance",
                    "ParameterApplyStatus": "pending-reboot"
                }
            ]
        }
```
If I remember correctly, the community provider handled this by actually restarting the cluster / instance,  but I have to test the upbound one.

Reply:
@nabuskey Are saying that the community provider does not wait for the maintenance window to restart the DB? If so, then we should open an issue in the crossplane repo.

Reply:
I did some testing and found that neither provider restarts the instance when a static parameter is updated. 

I think it's fine for the provider to restart if the apply immediately option is specified. The concept of maintenance window and all that do not really play well with GitOps imo.